src/Columns/ColumnUnique.h

```diff
@@ -361,6 +361,17 @@ size_t ColumnUnique<ColumnType>::uniqueInsert(const Field & x)
 
     FieldVisitorGetData visitor;
     applyVisitor(visitor, x);
+    if (valuesHaveFixedSize())
```

Comment:
Minor: while at it, could you please also put `FieldVisitorGetData` from an anonymous namespace into namespace `impl` or `detail` (l. 320)? I am getting some local warnings about it.

Reply:
It would be a bit nicer if the `if constexpr` was the outer if.

Reply:
AFAIS, the same fix is needed for all callers of `uniqueInsertData()` in this class.